packages/core-data/src/types/README.md

```diff
@@ -0,0 +1,211 @@
+    "rendered": "\n<p>Welcome to WordPress. This is your first post. Edit or delete it, then start writing!</p>\n"
+  },
+  "title": {
+    "raw": "Hello world!",
+    "rendered": "Hello world!"
+  }
+  // other fields
+}
+```
+
+And, finally, requesting `/wp/v2/posts/1?context=embed` yields:
+
+```js
+{
+  // Note content is missing
+  "title": {
+    "rendered": "Hello world!"
+  }
+  // other fields
+}
+```
+
+These contexts are supported by the core-data resolvers like `getEntityRecord()` and `getEntityRecords()` to retrieve the appropriate "flavor" of the data.
+
+The types describing different entity records must thus be aware of the relevant API context. This is implemented using the `Context` type parameter. For example, the implementation of the `Post` type resembles the following snippet:
+
+```ts
+import { OmitNevers } from "./helpers";
+
+interface BasePost<C extends Context> {
+	/**
+	 * A named status for the post.
+	 */
+	status: ContextualField< PostStatus, 'view' | 'edit', C >;
+
+	// ... other fields ...
+}
+
+export type Post<C extends Context> = OmitNevers<BasePost<C>>;
+```
+
+The `status` field of the `BasePost` is either a `PostStatus` when the Context is `view` or `edit`, or it's `never` when the Context is `embed`.
+
+The derived type `Post` is stripped of all the `never` fields, and so `Post<'embed'>` does not have the `status` type at all.
+
+### Static type checks for *edited* entity records, where certain fields become strings instead of objects.
+
+When the `post` is retrieved using `getEntityRecord`, its `content` field is an object:
+
+```js
+const post = wp.data.select('core').getEntityRecord( 'postType', 'post', 1, { context: 'view' } )
+// `post.content` is an object with two fields: protected and rendered
+```
+
+The block markup stored in `content` can only be rendered on the server so the REST API exposes both the raw markup and the rendered version. For example, `content.rendered` could used as a visual preview, and `content.raw` could be used to populate the code editor.
+
+The `content` field may still be updated via Javascript, but the updated content cannot be properly rendered without requesting the server. Therefore, it doesn't make sense to keep track the `rendered` when editing. The only remaining part is `raw`, and since that's a string, the entire edited `content` can be expressed a string.
+
+And this is exactly what you'll see after retrieving the editable version of the record using `getEditedEntityRecord`:
+
+```js
+const post = wp.data.select('core').getEditedEntityRecord( 'postType', 'post', 1 );
+// `post.content` is a string
+```
+
+## Helpers
+
+@TODO – perhaps we could auto generate that from the docstrings in common.ts (to become helpers.ts?)
+
+### Context
+
+The REST API context parameter.
+
+### ContextualField
+
+A wrapper that turns fields unavailable in the current REST API contexts into the type `never`. It is typically paired with OmitNevers.
```

Comment:
Again, let's focus on the outcome and goal instead of the implementation. We may change how it works, but the point is that `ContextualField` makes your type available in the given contexts and that field will be absent from the object when in a different context.